nrf_rpc/nrf_rpc.c

```diff
@@ -465,6 +467,27 @@ static int transport_init(nrf_rpc_tr_receive_handler_t receive_cb)
 	return err;
 }
 
+static int transport_init_all(nrf_rpc_tr_receive_handler_t receive_cb)
+{
+	void *iter;
+	const struct nrf_rpc_group *group;
+
+	for (NRF_RPC_AUTO_ARR_FOR(iter, group, &nrf_rpc_groups_array,
+				 const struct nrf_rpc_group)) {
+
+		transport_init_single(receive_cb, group);
+	}
+
+	/* Group initialization errors are not propagated to the caller. */
+	return groups_init_event_wait();
+}
+
+static void default_err_handler(const struct nrf_rpc_err_report *report)
```

Comment:
IMHO, this is a bit inconsistent that when you want to have groups auto-initialized then you can register a custom error handler and if you want to init groups manually then you have to use the default error handler.

Would it make sense to create a public API such as `nrf_rpc_init_core` which would only initialize the nRF PRC without any groups (and which would take the error handler). `nrf_rpc_init_group` would then validate that the core has already been initialized.